tools/binlog/binlog-slave-client.md

```diff
@@ -0,0 +1,147 @@
+---
+title: Binlog Slave Client 用户文档
+category: tools
+---
+
+# Binlog Slave Client 用户文档
+
+目前 Drainer 提供了多种输出方式，包括 MySQL、TiDB、TheFlash、pb 格式文件等。但是用户往往有一些自定义的需求，比如输出到 Elasticsearch、Hive 等，这些需求 Drainer 现在还没有实现，因此 Drainer 增加了输出到 Kafka 的功能，将 binlog 数据解析后按一定的格式再输出到 Kafka 中，用户编写代码从 Kafka 中读出数据再进行处理。
+
+## 配置 Drainer
+
+修改 Drainer 的配置文件，设置输出为 Kafka，相关配置如下：
+
+```
+[syncer]
+db-type = "kafka"
+
+[syncer.to]
+# Kafka 地址
+kafka-addrs = "127.0.0.1:9092"
+# Kafka 版本号
+kafka-version = "0.8.2.0"
+```
+
+## 自定义开发
+
+### 数据格式
+
+首先需要了解 Drainer 写入到 Kafka 中的数据格式：
+
+```
+// Column 保存列的数据，针对数据的类型，保存在对应的变量中
+message Column {
+  // 数据是否为 null
+  optional bool is_null = 1 [ default = false ];
+  // 保存 int 类型的数据
+  optional int64 int64_value = 2;
+  // 保存 uint、enum, set 类型的数据
+  optional uint64 uint64_value = 3;
+  // 保存 float、double 类型的数据
+  optional double double_value = 4;
+  // 保存 bit、blob、binary、json 类型的数据
+  optional bytes bytes_value = 5;
+  // 保存 date、time、decimal、text、char 类型的数据
+  optional string string_value = 6;
+}
+
+// ColumnInfo 保存列的信息，包括列名、类型、是否为主键
+message ColumnInfo {
+  optional string name = 1 [ (gogoproto.nullable) = false ];
+  // MySQL 中小写的列字段类型
+  // https://dev.mysql.com/doc/refman/8.0/en/data-types.html
+  // numeric 类型：int bigint smallint tinyint float double decimal bit
+  // string 类型：text longtext mediumtext char tinytext varchar
+  // blob longblog mediumblog binary tinyblob varbinary
```

Comment:
we have fixed the typo for the source code in https://github.com/pingcap/tidb-tools/pull/232.
you can update the doc to match it.

Reply:
Fixed in https://github.com/pingcap/docs-cn/pull/1264/commits/da3bc527f8a1d468b3c07bb13cdd518207b92815

Reply:
`longblog` still should be fixed.:hushed: